server/app/src/main/java/com/google/heartrate/wearos/app/gatt/heartrate/characteristics/HeartRateMeasurementCharacteristic.java

```diff
@@ -0,0 +1,269 @@
+                    EXPENDED_ENERGY_BASE_OFFSET);
+        } else {
+
+            /* Flags(UInt8) + HR(UInt16) + EE(UInt16) */
+            setValue(new byte[5]);
+
+            /* EE present and HR UInt16 format => Flags = 0b00001001 */
+            setIntValue((byte) (HEART_RATE_MEASUREMENT_UINT16_FLAG | EXPENDED_ENERGY_FLAG),
+                    BluetoothGattCharacteristic.FORMAT_UINT8,
+                    FLAGS_OFFSET);
+
+            setIntValue(heartRateMeasurement,
+                    BluetoothGattCharacteristic.FORMAT_UINT16,
+                    HEART_RATE_MEASUREMENT_OFFSET);
+
+            /* HR is in UInt16 format => EE offset is shifted */
+            setIntValue(expendedEnergy,
+                    BluetoothGattCharacteristic.FORMAT_UINT16,
+                    EXPENDED_ENERGY_SHIFTED_OFFSET);
+        }
+    }
+
+    /**
+     * Set given Heart Rate Measurement value to the characteristic.
+     * <p>
+     * Flags 8bit format 0b00000000 from right to left:
+     * <br>Heart Rate Format (0) -> UInt8 Format, (1) -> UInt16 Format
+     * <br>Sensor Contact Status (00) -> Not Supported
+     * <br>Energy Expended (0) -> Field not Present
+     * <br>RR-Interval (0) -> Field not presented
+     * <br>Unused (000)
+     *
+     * @param heartRateMeasurement (bpm) heart rate measurement value in UInt8/UInt16 format
+     * @throws GattException in case of wrong arguments format or offset for heart rate characteristic values
+     */
+    public void setHeartRateCharacteristicValue(int heartRateMeasurement) throws GattException {
+        Log.i(TAG, String.format("Set the heart measurements value %d", heartRateMeasurement));
+
+        FormatUtils.assertIsUInt16(heartRateMeasurement);
+
+        if (FormatUtils.isInUInt8Range(heartRateMeasurement)) {
+
+            /* Flags(UInt8) + HR(UInt8) */
+            setValue(new byte[2]);
+
+            /* EE not present and HR UInt8 format => Flags = 0b00000000 */
+            setIntValue(HEART_RATE_MEASUREMENT_UINT8_FLAG,
+                    BluetoothGattCharacteristic.FORMAT_UINT8,
+                    FLAGS_OFFSET);
+
+            setIntValue(heartRateMeasurement,
+                    BluetoothGattCharacteristic.FORMAT_UINT8,
+                    HEART_RATE_MEASUREMENT_OFFSET);
+
+        } else {
+
+            /* Flags(UInt8) + HR(UInt16) */
+            setValue(new byte[3]);
+
+            /* EE not present and HR UInt16 format => Flags = 0b00000001 */
+            setIntValue(HEART_RATE_MEASUREMENT_UINT16_FLAG,
+                    BluetoothGattCharacteristic.FORMAT_UINT8,
+                    FLAGS_OFFSET);
+
+            setIntValue(heartRateMeasurement,
+                    BluetoothGattCharacteristic.FORMAT_UINT16,
+                    HEART_RATE_MEASUREMENT_OFFSET);
+        }
+    }
+
+    /**
+     * Assert Expended Energy is supported.
+     *
+     * @throws GattException if not supported
+     */
+    private void assertExpandedEnergyIsPresented() throws GattException {
```

Comment:
nit: typo expanded -> expended

Reply:
Fixed in [commit](https://github.com/googleinterns/heartrate-bt-wear/pull/1/commits/f81955e6ee20af370c211fd96123abd538b599e1)